File: MediCareWebApi/Models/DoctorsAvailability.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MediCare.Models
{
	public class DoctorsAvailability : Entity
	{
		[Required]
		public DateTime From { get; set; }

		public DateTime? To { get; set; }

		[Required]
		public int DoctorsUserId { get; set; }

		[ForeignKey("DoctorsUserId")]
		public Doctor Doctor { get; set; }
	}
}

File: MediCareWebSpa/src/app/DTOs/models/doctors-availability.ts
import { Doctor } from "./doctor";

export class DoctorsAvailability {
  id: number;
  doctorsUserId: number;
  doctor: Doctor;
  from: Date;
  to: Date;

  constructor(
    id: number,
    doctorsUserId: number,
    doctor: Doctor,
    from: Date,
    to: Date
  ) {
    this.id = id;
    this.doctorsUserId = doctorsUserId
    this.doctor = doctor;
    this.from = from;
    this.to = to;
  }
}
